immutablecollections/_immutableset.py

```diff
@@ -235,6 +236,12 @@ def __sub__(self, other: AbstractSet[Any]) -> "ImmutableSet[T]":
         """
         return self.difference(other)
 
+    def __rsub__(self, other: AbstractSet[V]) -> "AbstractSet[V]":
+        """
+        Subtracts this set from a different set
+        """
+        return set(item for item in other if item not in self)
```

Comment:
Even though the order is not necessarily deterministic (on non-CPython implementations of 3.6), we can still try and at least return a set with the type of `other`. Using the type as the constructor should be ok:
```suggestion
        return type(other)(item for item in other if item not in self)
```

Reply:
This is one of those annoying things about Python - it appears to be assumed by even the `set` code itself that any set has a constructor which accepts only the items involved, but as far as I know it is not documented as part of the set interface.  Given that weirdness, I'd rather just return a regular set and make no guarantee in the typing beyond returning `AbstractSet`.

Reply:
Ok. It would have been a band-aid until the minimum Python version required is 3.7, then an `ImmutableSet` can be returned.